ipaserver/install/krbinstance.py

```diff
@@ -47,6 +49,86 @@
 from ipaplatform.tasks import tasks
 from ipaplatform.paths import paths
 
+PKINIT_ENABLED = 'pkinitEnabled'
+PKINIT_DISABLED = 'pkinitDisabled'
+
+
+def get_pkinit_request_ca():
+    """
+    Return the certmonger CA name which is serving the PKINIT certificate
+    request. If the certificate is not tracked by Certmonger, return None
+    """
+    pkinit_request_id = certmonger.get_request_id(
+        {'cert-file': paths.KDC_CERT})
+
+    if pkinit_request_id is None:
+        return
+
+    return certmonger.get_request_value(pkinit_request_id, 'ca-name')
+
+
+def get_local_pkinit_state():
+    """
+    Get the PKINIT feature status locally from the presence of KDC certificate
+    and certmonger tracking status
+    :returns: PKINIT status (local, external, full) or None if the feature is
+        not configured
+    """
+
+    if not os.path.exists(paths.KDC_CERT):
+        return
+
+    pkinit_request_ca = get_pkinit_request_ca()
+
+    if pkinit_request_ca is None or pkinit_request_ca == "IPA":
+        return PKINIT_ENABLED
+    elif pkinit_request_ca == "SelfSign":
+        return PKINIT_DISABLED
```

Comment:
@HonzaCholasta true, the selfsign/absent PKINIT information is actually required only locally during upgrade (that can be inferred from the presence of KDC certificate and its managing CA), server/replica install make decisions only based on provided options and state of topology.

It can thus be simplified further by reporting only 'pkinitEnabled' in LDAP and handling selfsign/absent only locally during upgrade. @simo5 is that ok with you?